src/wp-includes/default-filters.php

```diff
@@ -195,15 +195,15 @@
 add_filter( 'the_content', 'wpautop' );
 add_filter( 'the_content', 'shortcode_unautop' );
 add_filter( 'the_content', 'prepend_attachment' );
-add_filter( 'the_content', 'wp_filter_content_tags' );
+add_filter( 'the_content', 'wp_filter_content_tags', 12 ); // Runs after do_shortcode().
```

Comment:
Regarding the priority change, also see https://core.trac.wordpress.org/ticket/58853#comment:2 for the research I did prior to the PR.

I've updated the order for better readability in https://github.com/WordPress/wordpress-develop/pull/4973/commits/82dc4f3f1c501c6540cd7d7a4341b8261c4a5d15. I've also colocated the `the_content` filter for `do_shortcode` with all the other `the_content` ones, whereas before it was completely separated without a good reason (it had a "shortcodes" comment above, but all other `do_shortcode` callbacks were not there, so that really didn't make much sense anymore).